Define possible DP guarantees, and their respective composition theor…

junyechen1996 opened the pull request:
…ems for DP budgeting

Reviewer Note: stacked PR on #2

junyechen1996 committed:
Define possible DP guarantees, and their respective composition theorems for DP budgeting

wangshan committed:
Merge pull request #2 from wangshan/junyec/paper-citations

Paper citations that we will need

cjpatton commented on draft-wang-ppm-differential-privacy.md:
```diff
@@ -236,10 +236,61 @@ the gap between actual number of Clients and minimum batch size.
+
+* Composition theorems that apply to the DP guarantee.
+
+### Pure ϵ-DP, or (ϵ,δ)-approximate DP {#adp}
```
Unfortunately unicode characters are not supported by IETF's tooling: https://wangshan.github.io/draft-wang-ppm-differential-privacy/junyec/dp-budget/draft-wang-ppm-differential-privacy.html#name-pure-1013-dp-or-1013948-app

cjpatton replied:
I'd suggest names like `EPSILON` and `DELTA`.

junyechen1996 replied:
Fixed all the unicode characters.

cjpatton commented on draft-wang-ppm-differential-privacy.md:
```diff
@@ -236,10 +236,61 @@ the gap between actual number of Clients and minimum batch size.
+* Federated machine learning applications are usually composed of multiple
+  iterations.
+
+* {{MJTBCCDFGHJKLLMPPPPPRSWZ22}} describes an interactive approach of building
```
In my opinion this is too long of an authors list to be useful as a reference. I'd suggest truncating to the top three or four authors (and make sure to be consistent throughout the doc.)

cjpatton replied:
E.g., {{MJTBp22}}, where "p" stands for "plus"

junyechen1996 replied:
Done.

wangshan reviewed (approved):
LG,

One thing we can discuss later: is budget a MUST for client or SHOULD , if it’s a must then do we need to spell out exactly how it’s done?

cjpatton commented on draft-wang-ppm-differential-privacy.md:
```diff
@@ -236,10 +236,61 @@ the gap between actual number of Clients and minimum batch size.
+Many applications need to query the Client data multiple times, for example:
+
+* Federated machine learning applications are usually composed of multiple
+  iterations.
```
Can we reframe this in the DAP context? "Federated machine learning applications require multiple aggregates to be computed over the same underlying data."

junyechen1996 replied:
Rephrased it, and added "with different machine learning model parameters".

cjpatton commented on draft-wang-ppm-differential-privacy.md:
```diff
@@ -236,10 +236,61 @@ the gap between actual number of Clients and minimum batch size.
+
+It’s hard for Aggregator(s) to keep track of the privacy budget over time,
+because different Clients can participate in different data collection tasks,
+and only Clients know when their data is queried. Therefore, Clients MUST
```
I don't think a normative "MUST" isn't appropriate here since we're not talking about a concrete enforcement mechanism yet. I would just write "must".

cjpatton commented on draft-wang-ppm-differential-privacy.md:
```diff
@@ -236,10 +236,61 @@ the gap between actual number of Clients and minimum batch size.
+can be found in Definition 2.4 of {{DR14}}.
+
+One can compose multiple (ϵ,δ)-approximate DP guarantees, per Theorem 3.4
+of {{KOV15}}.
```
Wrap lines at 80 characters

junyechen1996 replied:
This line is ok, but I changed a couple of other places that have more than 80 characters.

cjpatton reviewed (commented):
In addition to pointing to formal definitions of DP notions, we should provide informal definitions that help develop the reader's intuition.

junyechen1996 committed:
comments

- provide intuitions in each DP guarantee
- wrap some lines over 80 characters
- rename unicode characters to uppercase words

junyechen1996 commented:
> In addition to pointing to formal definitions of DP notions, we should provide informal definitions that help develop the reader's intuition.

I tried to provide some intuitions for each DP guarantees. Let me know if this is better.

cjpatton commented:
> > In addition to pointing to formal definitions of DP notions, we should provide informal definitions that help develop the reader's intuition.
> 
> I tried to provide some intuitions for each DP guarantees. Let me know if this is better.

Much better, but what I want us to be able to do is answer the following question: Why would one choose some notion of DP over another? Are there certain restrictions/requirements that prevent us from obtaining one but not anohter?

cjpatton commented:
Oops :( my bad I forgot to check the base before merging.

cjpatton commented:
Merged into main.

junyechen1996 commented:
Thanks! I rebased to main, but also forgot to change the target branch.

junyechen1996 commented:
> but what I want us to be able to do is answer the following question: Why would one choose some notion of DP over another? Are there certain restrictions/requirements that prevent us from obtaining one but not anohter?

Good point, we can do better. I filed #6 to track this
